### PR TITLE
Update Snowman project for 8.0

### PR DESCRIPTION
**Pull request by bwreid**
- Reorganize project structure.
- Update package.json with script.
- Update readme and rubric.
- Update snowman.js file to have commented code.

### CODE REVIEW

File: rubric.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+### Git
+
+- **Exemplary:** Multiple commits have been created, showing progress towards creating the game. Each commit message has an excellent and descriptive commit message.
+
+- **Proficient:** Multiple commits have been created, showing progress towards creating the game. Each commit message has a sensible commit message.
+
+- **Developing:** Two or three commits have been made, showing progress towards creating the game. Alternatively, the commit messages are not descriptive.
+
+- **Novice:** Few commits have been made and/or the commit messages are not descriptive at all.
+
```
Comment (mikeboyle) [inline rubric.md:26]: I'm of two minds here.
In one way, I love this standard for git and we should be enforcing it somehow.

In another way, if a fellow scores `developing` or `novice` here, there is not a realistic way for them to fix the mistake and resubmit for a higher score. They could reorganize the commits with interactive rebase but that would be quite a tricky project, and if they could use interactive rebase they would be deserving of the exemplary score.

Am I overlooking an easy way of fixing poor git habits in the submission? And if not, do you have any advice for how to handle module gate situations where they need these points to be considered passing the gate?

File: src/snowman.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+*/
+function getRandomWord() {
+  const index = Math.floor(Math.random() * dictionary.length);
+  return dictionary[index];
+}
+
+/*
+  This function will run your game. Everything you want to happen in your game should happen inside of here.
+
+  You can still create other, smaller functions that will be called inside of the `run()` function.
```
Comment (mikeboyle) [inline src/snowman.js:21]: Suggest rewording this to make it less ambiguous and more direct (helper functions are a rubric area):
```
You should still define other, smaller functions outside of the `run()` function that have a single specific purpose, such as getting user input or checking if a guess is correct. You can then call these helper functions from inside the `run()` function.
``

File: src/snowman.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+  Once you understand the code below, you may remove the comments if you like.
+*/
```
Comment (mikeboyle) [inline src/snowman.js:24]: Thinking back on the challenges we've had with this project, I think we should give the fellows a bit more in the way of starter code to support these areas:
1. `Initializing, managing, and reading game state`: This is a software design skill that I don't think we will have taught or given them an opportunity to practice before. It's very hard if you've never done it before. I would be okay with more starter code that creates a state object in memory, maybe even with some important keys in it, and explain in a comment (and in class) how code executed from `run()` would interact with the state.
2. `Game logic and helper functions`: I believe this is also going to be a new area, where the logic needed to run the game is found not in one function but rather spread out over several functions like `isGameOver()`, `isGuessCorrect()`, `isGuessValid()`, `isGameWon()` etc. This is also kind of a software design challenge that we will not have exposed them to. Is it reasonable to give them some kind of helper functions ... just the function signature with jsdoc description of what they do?
3. `Command line app things`: In particular the way in which a `while()` loop (maybe more than one while loop) is used to allow repeated player turns, and when/how to break the while loop (when the game is over, when the user wants to quit, etc.)

If we don't want to give these things as starter code, I think inevitably we'll end up having to support with the same content in class, across multiple office hours, etc. It will probably save time and heartache either to give the starter code or to make time/content for some kind of `how to set up the project` lesson in class (or video).

**Review by mikeboyle — CHANGES_REQUESTED** [review]
Love the improvements here, especially making the starter code more comprehensible. This project was incredibly challenging for 7.0, not always in a good way, so I'm suggesting some ways to make the challenge a little less steep.

File: rubric.md (hunk 1 above)
Comment (bwreid) [reply rubric.md:26]: Fair point. I think, as opposed to resubmitting a PR, remedying a low score on this might be something like simply sending a message that describes:
* The kind of commits they _should_ have been making.
* The kind of messages they _should_ have included.

This _does not_ fix the "habit" part of Git, but might satisfy the "knowledge" part. Does that seem sufficient?

Comment (mikeboyle) [reply rubric.md:26]: Yes, I think that can work. They can submit a comment on Canvas.

File: src/snowman.js (hunk 3 above)
Comment (mikeboyle) [reply src/snowman.js:24]: update on this: we're going to do some support content for this (either videos or in class or both) so there is less need to have this built out starter code or support commenting here.

**Review by mikeboyle — APPROVED** [review]
